dev/detect and remove outliers.qmd

```diff
@@ -0,0 +1,276 @@
+---
+format:
+  gfm: default
+  html: default
+---
+
+# Function `remove_outliers()`: detect and remove outliers based on speed
+
+This is a function developed to identify and remove outliers from GPS data using
+speed as the detecting metric. Speed is a good metric for outlier detection as
+it expresses distance covered given elapsed time, therefore highlighting
+instances of erroneous movement caused by sensor malfunctions. Here we have a
+close look at the function developed to perform this task.
+
+```{r}
+#| include: false
+#| warning: false
+#| message: false
+
+library(dplyr)
+library(ggplot2)
+library(move2)
+library(sf)
+library(units)
+
+source("../RFunction.R")
+source(file.path("../src/common/logger.R"))
+
+input1 <- readRDS("../data/raw/input1.rds")
+input2 <- readRDS("../data/raw/input2.rds")
+
+```
+
+
+Let's peak at the function's code.
+```{r}
+remove_outliers
+```
+
+
+Generate mock base data.
+```{r}
+n <- 10
+dt <- data.frame(
+  x = seq(30.01, 31.02, length.out = n), 
+  y = seq(2.01, 3.02, length.out = n),
+  time = seq.POSIXt(
+    as.POSIXct("2023-01-01 00:00:00 UTC"),
+    as.POSIXct("2023-01-01 01:30:00 UTC"), 
+    length.out = n
+  ), 
+  track = "a"
+)
+
+dt_mv2_base <- mt_as_move2(
+  dt,
+  coords = c("x", "y"), 
+  time_column = "time",
+  track_id_column = "track") |> 
+  sf::st_set_crs(4326L)
+
+dt_mv2_base
+```
+
+
+## Application to two consecutive outliers
+
+Add two outliers to the 2nd and 3rd positions and ad speeds (km/h).
```

Comment:
missing a "d" (add)

Reply:
Typo-alert!

Fixed in https://github.com/dmpstats/Standardise_Formats_and_Calculate_Basic_Statistics/pull/6/commits/371ad170ca546aa4b7252085a3814994bb328a19